.circleci/config.yml

```diff
@@ -5,15 +5,15 @@ orbs:
 
 executors:
   ubuntu:
-    working_directory: ~/go/src/github.com/mattermost/
+    working_directory: ~/go/mattermost/
     machine:
       image: "ubuntu-1604:201903-01"
     environment:
       COMPOSE_PROJECT_NAME: "circleci"
 
 jobs:
   setup:
-    working_directory: /go/src/github.com/mattermost/mattermost-server
+    working_directory: ~/go/mattermost/mattermost-server
```

Comment:
Could you elaborate a bit why we need to work under `~/go`? I could not find any references to `$GOPATH/bin` that require this.

Reply:
It is more about keeping things consistent. There is no requirement to keep it under `~/go`. But the `make package` target copies binaries from $GOPATH/bin to $DIST_PATH/bin. It seemed inconsistent to me, for us to move the repo directories totally out of GOPATH, but still use $GOPATH/bin for binary placement.

My plan is to send a future PR (by me or anybody else), where $GOPATH/bin is replaced by $GOBIN and completely remove ~/go and just work out of ~/mattermost. All in one PR. So that we do not fall in a half-baked state.

Reply:
Thanks for the explanation :+1: 

I'm fine with merging this as a first step :+1: